### PR TITLE
[model] feat: polish model engine

### PR DESCRIPTION
**Pull request by vermouth1992**
### What does this PR do?

1. Refactor config into more modular configs including fsdp/megatron optim and engine. Then reuse them in all other high level configs
2. Add loss_function to infer batch to cover validation of SFT. Unify model engine output to consist 1) model outputs; 2) loss; 3) metrics. Then add a general postprocess function
3. Modify registry system to include a model type and backend
4. Add dp_group to the args of loss function

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)

### CODE REVIEW

File: verl/workers/roles/utils/losses.py [hunk 1]
```diff
@@ -19,14 +19,14 @@
 from verl.workers.config import ActorConfig
 
 
-def sft_loss(config: ActorConfig, model_output, data):
+def sft_loss(config: ActorConfig, model_output, data, dp_group=None):
```
Comment (ETOgaosion) [inline verl/workers/roles/utils/losses.py:22]: What's this field for?

**Review by ETOgaosion — APPROVED** [review]

File: verl/workers/roles/utils/losses.py (hunk 1 above)
Comment (vermouth1992) [reply verl/workers/roles/utils/losses.py:22]: Sometimes, we need to allgather from dp group to calculate loss and balance

File: verl/trainer/config/_generated_ppo_trainer.yaml [hunk 2]
```diff
@@ -42,12 +42,16 @@ actor_rollout_ref:
       load_contents: ${.save_contents}
       async_save: false
     optim:
-      lr: 1.0e-06
+      lr: 0.001
```
Comment (ETOgaosion) [inline verl/trainer/config/_generated_ppo_trainer.yaml]: A breaking change? And the lr is the same with critic now.

File: verl/workers/roles/utils/losses.py (hunk 1 above)
Comment (vermouth1992) [reply verl/workers/roles/utils/losses.py:22]: Just like this: https://github.com/volcengine/verl/blob/main/verl/trainer/fsdp_sft_trainer.py#L442

File: verl/trainer/config/_generated_ppo_trainer.yaml (hunk 2 above)
Comment (vermouth1992) [reply verl/trainer/config/_generated_ppo_trainer.yaml]: Yes. Since they inherit the same optim config, there is no way to make them different.

Comment (vermouth1992) [reply verl/trainer/config/_generated_ppo_trainer.yaml]: Most of the time, the scripts would override the learning rate
